### PR TITLE
Ports the Ore Redemption Machine to TGUI

### PR DESCRIPTION
**Pull request by Expletive**
:cl:
add: The Ore Redemption Machine has been ported to TGUI. New features include the addition of "Release All" and "Smelt All" buttons.
add: The Ore Redemption Machine now be loaded with sheets of material.
add: The Ore Redemption Machine can now 'smelt' Reinforced Glass.
/:cl:

Everything needs to be ported to TGUI, right? This one seemed simple enough, but it took me a while to make it stop crashing the entire server. Now it should be fine though. 
And this PR doesn't have any merge conflicts. Yet.

Edit: After looking at some stuff, I've managed to make the thing look like this:
Edit: Now with text fields! If you enter a number in the field, you can now release that many sheets by pressing the release button. This means the popup will only appear if you don't enter a number.
ormtext

### CODE REVIEW

**Comment by ShadowDeath6** [comment]
Add another thing to the list of things that uses lag interface

**Comment by Profakos** [comment]
Absolutely beautiful.

**Comment by XDTM** [comment]
>Release All

🎉

**Comment by xxalpha** [comment]
Stop porting shit to tgui until tgui is fixed.

**Comment by Kromgar** [comment]
Release all

YES YES YES YES YESSSSSSSS

File: code/modules/mining/machine_redemption.dm [hunk 1]
```diff
@@ -205,142 +205,141 @@
...
+	var/list/data = list()
+	data["unclaimedPoints"] = points
 	if(inserted_id)
-		dat += "You have [inserted_id.mining_points] mining points collected. <A href='?src=\ref[src];eject_id=1'>Eject ID.</A><br>"
-		dat += "<A href='?src=\ref[src];claim=1'>Claim points.</A><br><br>"
+		data["hasID"] = TRUE
+		data["claimedPoints"] = inserted_id.mining_points
 	else
-		dat += "No ID inserted.  <A href='?src=\ref[src];insert_id=1'>Insert ID.</A><br><br>"
+		data["hasID"] = FALSE
```
Comment (lzimann) [inline code/modules/mining/machine_redemption.dm]: This is not needed. Not setting hasID will already return false in the template.

File: code/modules/mining/machine_redemption.dm [hunk 2]
```diff
@@ -205,142 +205,141 @@
...
-/obj/machinery/mineral/ore_redemption/proc/get_ore_values()
-	var/dat = "<table border='0' width='300'>"
-	for(var/ore in ore_values)
-		var/value = ore_values[ore]
-		dat += "<tr><td>[capitalize(ore)]</td><td>[value * point_upgrade]</td></tr>"
-	dat += "</table>"
-	return dat
-
-/obj/machinery/mineral/ore_redemption/Topic(href, href_list)
+		data["hasDisk"] = FALSE
```
Comment (lzimann) [inline code/modules/mining/machine_redemption.dm]: Same here.

**Comment by Repukan** [comment]
God bless you, child

**Comment by RemieRichards** [comment]
@Expletive You did it! you made the buttons good! :clap:

**Comment by GunHog** [comment]
Did you read a guide somewhere for this? I would like to read it as well!

**Comment by ChangelingRain** [comment]
Wait that's a thing you can do

**Comment by Penguaro** [comment]
I'm concerned abput release all. I see some scientist just hitting that for a few things they need and leaving the rest on the ground which can't bd reinserted.

**Comment by Repukan** [comment]
Nah they usually take everything like the material starved sluts they are.

**Comment by Niknakflak** [comment]
Just another thing players can't use ingame because TGUI sucks cock

File: code/modules/mining/machine_redemption.dm [hunk 3]
```diff
@@ -205,142 +211,148 @@
...
-
-/obj/machinery/mineral/ore_redemption/Topic(href, href_list)
+		data["hasDisk"] = TRUE
+		if(inserted_disk.blueprints.len)
+			var/index = 1
+			for (var/datum/design/thisdesign in inserted_disk.blueprints)
+				if(thisdesign)
+					data["diskDesigns"] += list(list("name" = thisdesign.name, "index" = index, "canupload" = thisdesign.build_type&SMELTER))
+				index++
+				CHECK_TICK
```
Comment (optimumtact) [inline code/modules/mining/machine_redemption.dm]: [n]ice

File: code/modules/mining/machine_redemption.dm [hunk 4]
```diff
@@ -205,142 +211,148 @@
...
+			var/alloy_id = params["id"]
+			var/datum/design/alloy = files.FindDesignByID(alloy_id)
+			if((check_access(inserted_id) || allowed(usr)) && alloy)
+				var/smelt_amount = can_smelt_alloy(alloy)
+				while(smelt_amount > 0)
+					materials.use_amount(alloy.materials)
+					smelt_amount--
+					var/output = new alloy.build_path(src)
+					unload_mineral(output)
+					CHECK_TICK
```
Comment (optimumtact) [inline code/modules/mining/machine_redemption.dm]: nice

**Pull request merged**